src/App.tsx

```diff
@@ -1,204 +1,72 @@
+/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
 import { USER_ID } from './api/todos';
+import * as servisesTodos from './api/todos';
+import { Todo } from './types/Todo';
+import { AddTodo } from './components/AddTodo/AddTodo';
+import { TodoList } from './components/TodoList/TodoList';
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
+import { FooterFiltering } from './components/FooterFiltering/FooterFiltering';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState('All');
```

Comment:
Create a union type for it and use it as generic here
```suggestion
  const [filter, setFilter] = useState<Filter>('All');
```